File: C#.Net/Tests/TutorialTests/Career/PageObjects/Pages/JobListingPage.cs
using Epam.JDI.Web.Attributes;
using Epam.JDI.Web.Selenium.Elements.Complex.table;
using Epam.JDI.Web.Selenium.Elements.Complex.table.interfaces;
using Epam.JDI.Web.Selenium.Elements.Composite;
using OpenQA.Selenium;
using System;
using System.Linq;

namespace Epam.Tests.TutorialTests.Career.PageObjects
{
    public class JobListingPage : WebPage
    {
        [FindBy(ClassName = "search-result-list")]
        public ITable jobsList = (ITable)new Table(null,
            By.XPath(".//li[%s]//div"),
            By.XPath(".//li//div[%s]"))
            .HasColumnHeaders(Enum.GetNames(typeof(JobListHeaders)).ToList());

        public void getJobRowByName(string jobName)
        {
            // it is java
            //MapArray<String, ICell> row = jobsList.row(jobName, column(JOB_NAME.toString()));
            //row.get(APPLY.toString()).select();
        }

    }
}

File: Java/Tests/jdi-uitest-tutorialtests/src/main/java/com/epam/jdi/site/epam/pages/JobListingPage.java
package com.epam.jdi.site.epam.pages;

import com.epam.commons.map.MapArray;
import com.epam.jdi.entities.Vacancy;
import com.epam.jdi.site.epam.sections.VacancyRow;
import com.epam.jdi.uitests.core.interfaces.complex.tables.interfaces.ICell;
import com.epam.jdi.uitests.web.selenium.elements.complex.Elements;
import com.epam.jdi.uitests.web.selenium.elements.complex.table.EntityTable;
import com.epam.jdi.uitests.web.selenium.elements.complex.table.Table;
import com.epam.jdi.uitests.web.selenium.elements.composite.WebPage;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.objects.JTable;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.simple.Css;
import org.openqa.selenium.support.FindBy;

import java.util.List;

import static com.epam.commons.LinqUtils.where;

/**
 * Created by Roman_Iovlev on 10/22/2015.
 */
public class JobListingPage extends WebPage {

    @JTable(root = @FindBy(css = ".search-result__list"),
            row = @FindBy(xpath = ".//article[%s]/*"),
            column = @FindBy(xpath = ".//article/*[%s]"),
            header = {"name", "description", "applyShareCell"})
    public EntityTable<Vacancy, VacancyRow> jobsAsData;

    @JTable(root = @FindBy(css = ".search-result__list"),
            row = @FindBy(xpath = ".//article[%s]/*"),
            column = @FindBy(xpath = ".//article/*[%s]"),
            header = {"name", "description", "applyShareCell"})
    public Table jobs;

    @Css(".search-result__list>.search-result__item")
    public Elements<VacancyRow> jobsList;

    public void getJobRowByName(String jobName) {
        List<VacancyRow> result = getRowsByName(jobName);
    //    result.get(0).apply.click();
    }

public List<VacancyRow> getRowsByName(String jobName) {
    return where(jobsList, job -> job.name.getText().equals(jobName));
}

/*    public void getJobRow(String jobName) {
        MapArray<String, ICell> row = jobsList.row(jobName, column(name));
        row.get(apply.toString()).select();
    }*/
}
